src/logger.ts

```diff
@@ -38,32 +38,85 @@ export const debug_info = {
   interfaceDB_cleanTime: 0,
 }
 
-export async function saveInterfaceStat(): Promise<void> {
-  console.log(apiPerfLogData)
+/**
+ * Performs a batch insert operation into a database table.
+ * 
+ * @param db - The database instance to use for the operation.
+ * @param sql - The SQL query string for the insert operation.
+ * @param rawItems - An array of raw items to be mapped and inserted.
+ * @param mapConfig - An object describing how to map raw items to database fields.
+ */
+function batchInsert<T>(
+  sql: string, 
+  rawItems: T[], 
+  mapConfig: { [key: string]: (item: T) => any }
+) {
+  const stmt = db.prepare(sql)
+  const failedItems: T[] = []
+  const fields = Object.keys(mapConfig)
+  
+  // Create a db tx that groups multiple db operations into a single transaction
+  const transaction = db.transaction(() => {
+    for (let i = 0; i < rawItems.length; i++) {
+      const rawItem = rawItems[i]
+      try {
+        // Map and insert one item at a time for memory efficiency and flexibility
+        const mappedItem = fields.map(field => mapConfig[field](rawItem))
+        stmt.run(...mappedItem)
+      } catch (error) {
+        failedItems.push(rawItem)
+        // Don't throw here to allow the transaction to continue with other items
+      }
+    }
+  })
+
   try {
-    // eslint-disable-next-line prefer-const
-    let { api_name, tfinal, timestamp, nodeUrl, success, reason, hash } = apiPerfLogData[0]
-    // nodeUrl = nodeUrl ? nodeUrl : new URL(nodeUrl as string).hostname
-    let placeholders = `NULL, '${api_name}', '${tfinal}','${timestamp}', '${nodeUrl}', '${success}', '${reason}', '${hash}'`
-    let sql = 'INSERT INTO interface_stats VALUES (' + placeholders + ')'
-    for (let i = 1; i < apiPerfLogData.length; i++) {
-      // eslint-disable-next-line prefer-const
-      let { api_name, tfinal, timestamp, nodeUrl, success, reason, hash } = apiPerfLogData[i] // eslint-disable-line security/detect-object-injection
-
-      // nodeUrl = nodeUrl ? nodeUrl : new URL(nodeUrl as string).hostname
-      placeholders = `NULL, '${api_name}', '${tfinal}','${timestamp}', '${nodeUrl}', '${success}', '${reason}', '${hash}'`
-      sql = sql + `, (${placeholders})`
+    transaction()
+  } catch (error) {
+    console.error('Batch insert failed:', error)
+    const hashField = 'hash' in mapConfig ? 'hash' : 'txHash' in mapConfig ? 'txHash' : null
+    if (hashField && failedItems.length > 0) {
+      console.error('Failed item hashes:', failedItems.map(item => mapConfig[hashField](item)))
+    } else {
+      console.error('Number of failed items:', failedItems.length)
     }
+    throw error // Re-throw to allow custom handling in the calling function
+  }
+}
+
+export async function saveInterfaceStat(): Promise<void> {
+  console.log(apiPerfLogData)
```

Comment:
put behind verbose flag and have a descriptor of what is being logged: console.log('api perf log data', apiPerfLogData)